pkg/scheduler/factory/cache_comparer.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factory
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+	policy "k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/labels"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	v1beta1 "k8s.io/client-go/listers/policy/v1beta1"
+	"k8s.io/kubernetes/pkg/scheduler/schedulercache"
+)
+
+type cacheComparer struct {
+	nodeLister corelisters.NodeLister
+	podLister  corelisters.PodLister
+	pdbLister  v1beta1.PodDisruptionBudgetLister
+	cache      schedulercache.Cache
+
+	compareStrategy
+}
+
+func (c *cacheComparer) Compare() error {
+	glog.V(3).Info("cache comparer started")
+	defer glog.V(3).Info("cache comparer finished")
+
+	nodes, err := c.nodeLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	pods, err := c.podLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	pdbs, err := c.pdbLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	snapshot := c.cache.Snapshot()
+
+	if missed, redundant := c.CompareNodes(nodes, snapshot.Nodes); len(missed)+len(redundant) != 0 {
+		glog.Warningf("cache mismatch: missed nodes: %s; redundant nodes: %s", missed, redundant)
+	}
+
+	if missed, redundant := c.ComparePods(pods, snapshot.Nodes); len(missed)+len(redundant) != 0 {
```

Comment:
This part of the PR is not correct and will create false positives. Scheduler keeps assigned pods (pods with PodSpec.NodeName != "") in its cache. Unassigned pods live in scheduler's scheduling queue. For now it is OK to ignore unassigned pods. The code in its current form compares all pods against assigned ones in the scheduler cache. So, if there are pending pods in the cluster, we will see warning messages.

Reply:
I am not aware of that.. Let me fix it and rebase this PR.

Reply:
Updated @bsalamat